consumer/consumererror/consumererror.go

```diff
@@ -0,0 +1,165 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package consumererror // import "go.opentelemetry.io/collector/consumer/consumererror"
+
+import (
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"google.golang.org/grpc/status"
+)
+
+type Error struct {
+	error
+	retryable  bool
+	rejected   int
+	delay      time.Duration
+	hasTraces  bool
+	traces     ptrace.Traces
+	hasMetrics bool
+	metrics    pmetric.Metrics
+	hasLogs    bool
+	logs       plog.Logs
+	httpStatus int
+	grpcStatus *status.Status
+}
+
+var noDataCount = -1
+
+// ErrorOption allows annotating an Error with metadata.
+type ErrorOption func(error *Error)
+
+// NewConsumerError wraps an error that happened while consuming telemetry
+// and adds metadata onto it to be passed back up the pipeline.
+func NewConsumerError(origErr error, options ...ErrorOption) error {
+	err := Error{error: origErr, rejected: noDataCount}
+
+	cErr := Error{}
+	if errors.As(err, &cErr) {
```

Comment:
Always true since 2 lines above you make it an `Error`?

Reply:
Good catch. Fixed.